### DISCUSSION TITLE
Query regarding PyTorch tensors and NumPy

### DISCUSSION THREAD

**Discussion by VijayV28** [mrdbourke/pytorch-deep-learning#222] [Q&A] [answered]
Hey,

It was mentioned that when a Numpy array is converted to a Tensor, the tensor won't change even if the numpy is reassigned. Have a look at the code below:

**Converting a Numpy array to tensor**
```python
array_x = np.arange(1,8)
tensor_x = torch.from_numpy(array_x)
array_x, tensor_x
```

I tried reassigning a number in the array as shown below:
```python
array_x[0] = 9
array_x, tensor_x
```

The above change was reflected in the tensor:
***Output***
![image](https://user-images.githubusercontent.com/81868201/208443554-206548a2-8be9-4be2-a9fb-35b7a99e085c.png)

However, when I tried reassigning using the method given in the video, the change was not shown in the tensor:
```python
array_x = array_x + 1
array_x, tensor_x
```
***Output***
![image](https://user-images.githubusercontent.com/81868201/208444232-81b82cee-a23e-4fbc-88b7-d00888e89bf2.png)

Can anyone please explain why is this happening?

**Comment by AlessandroMiola** [comment, marked as answer]
I'd suggest to try to have a look at https://github.com/mrdbourke/pytorch-deep-learning/discussions/124.
Hope it may help!

**Reply by VijayV28** [reply]
Thank you very much.
